pkg/controller/storagecluster/reconcile.go

```diff
@@ -172,6 +176,29 @@ func (r *ReconcileStorageCluster) Reconcile(request reconcile.Request) (reconcil
 		if phaseErr != nil {
 			reqLogger.Error(phaseErr, "Failed to set PhaseDeleting")
 		}
+
+		// Check if the storagecluster is labelled for data deletion, if yes
+		// label the child resources for data deletion too before issuing delete
+		if v, found := instance.ObjectMeta.Labels[CleanupPolicyLabel]; found {
+			if v == CleanupPolicyDelete {
```

Comment:
Why not using the type instead of the new constant?

Reply:
In this line, we are referring to the ocs-operator cleanup policy and not rook's. I am not sure if it is a good idea to use the type from cephv1 although I am using the same string. 

Let me know if you would still prefer to use the type from ceph instead of a new constant local to ocs-operator.

Reply:
OCS should probably have its own type for this. It's just annoying to have to replicate all the things Rook does this way but it is what it is :).

At this point, I don't have a strong preference but I think a new type for OCS-op would make sense.

Reply:
Wait, why can't OCS reference the consts or types in Rook? OCS is already referencing the `cephv1` package.

Reply:
It can, just that it probably not a good idea to. The logic reads like this

```
if ocsCleanupLabel is set to ocsCleanupDeleteValue
then
   edit the CleanupSpec of Rook to cephv1.CleanupPolicyDelete
   edit the CleanupSpec of Noobaa //future
fi

```

It so happens that the ocsCleanupDeleteValue and the cephv1.CleanupPolicyDelete value both are "yes-really-destroy-data"

Reply:
Oh right, this really is an OCS setting so agreed we shouldn't reference rook's.